packages/fast-components-react-base/src/viewport-positioner/viewport-positioner.tsx

```diff
@@ -0,0 +1,1057 @@
+                    " ",
+                    get(this.props.managedClasses, "viewportPositioner__top", "")
+                );
+                classNames = classNames.concat(
+                    " ",
+                    get(
+                        this.props.managedClasses,
+                        "viewportPositioner__verticalFlipOutward",
+                        ""
+                    )
+                );
+                break;
+
+            case ViewportPositionerVerticalPositionLabel.centerTop:
+                classNames = classNames.concat(
+                    " ",
+                    get(this.props.managedClasses, "viewportPositioner__top", "")
+                );
+                classNames = classNames.concat(
+                    " ",
+                    get(
+                        this.props.managedClasses,
+                        "viewportPositioner__verticalFlipInward",
+                        ""
+                    )
+                );
+                break;
+
+            case ViewportPositionerVerticalPositionLabel.centerBottom:
+                classNames = classNames.concat(
+                    " ",
+                    get(this.props.managedClasses, "viewportPositioner__bottom", "")
+                );
+                classNames = classNames.concat(
+                    " ",
+                    get(
+                        this.props.managedClasses,
+                        "viewportPositioner__verticalFlipInward",
+                        ""
+                    )
+                );
+                break;
+
+            case ViewportPositionerVerticalPositionLabel.bottom:
+                classNames = classNames.concat(
+                    " ",
+                    get(this.props.managedClasses, "viewportPositioner__bottom", "")
+                );
+                classNames = classNames.concat(
+                    " ",
+                    get(
+                        this.props.managedClasses,
+                        "viewportPositioner__verticalFlipOutward",
+                        ""
+                    )
+                );
+                break;
+        }
+
+        return super.generateClassNames(classNames);
+    }
+
+    /**
+     *  gets the CSS classes to be programmatically applied to the component
+     */
+    private getPositioningStyles = (): {} => {
+        // determine if we should hide the positioner because we don't have data to position it yet
+        // (avoiding flicker)
+        let shouldHide: boolean = false;
+        if (this.state.disabled) {
+            if (
+                (this.props.disabled === undefined || this.props.disabled === false) &&
+                isNil(this.positionerRect) &&
+                this.getAnchorElement() !== null
+            ) {
+                shouldHide = true;
+            }
+        } else if (isNil(this.positionerRect) && !this.state.noOberverMode) {
+            shouldHide = true;
+        }
```

Comment:
seems like this could be simplified to:

```
const shouldHide: boolean = (this.state.disabled &&
    (
        (this.props.disabled === undefined || this.props.disabled === false) &&
                isNil(this.positionerRect) &&
                this.getAnchorElement() !== null
    )
) ||
(isNil(this.positionerRect) && !this.state.noOberverMode);
```

But as that is hard to read, you might put it in another method.

Reply:
yep